### ISSUE TITLE
aws_eks: lambda runtime for kubectl layer is node.js 12, which is deprecated and EOL

### ISSUE THREAD

**Issue by ctrepka** [aws/aws-cdk#22116]
### Describe the bug

Well, I received this message : AWS Lambda end of support for Node.js 12

    Hello,

    We are contacting you as we have identified that your AWS Account currently has one or more Lambda functions using Node.js 12 runtime.

    We are ending support for Node.js 12 in AWS Lambda. This follows Node.js 12 End-Of-Life (EOL) reached on April 30, 2022 [1].

    As described in the Lambda runtime support policy [2], end of support for language runtimes in Lambda happens in two stages. Starting November 14, 2022, Lambda will no longer apply security patches and other updates to the Node.js 12 runtime used by Lambda functions, and functions using Node.js 12 will no longer be eligible for technical support. In addition, you will no longer be able to create new Lambda functions using the Node.js 12 runtime. Starting December 14, 2022, you will no longer be able to update existing functions using the Node.js 12 runtime.

    We recommend that you upgrade your existing Node.js 12 functions to Node.js 16 before November 14, 2022.

    End of support does not impact function execution. Your functions will continue to run. However, they will be running on an unsupported runtime which is no longer maintained or patched by the AWS Lambda team.

    The following command shows how to use the AWS CLI [3] to list all functions in a specific region using Node.js 12. To find all such functions in your account, repeat this command for each region:

    aws lambda list-functions --function-version ALL --region us-east-1 --output text --query "Functions[?Runtime=='nodejs12.x'].FunctionArn"

    If you have any concerns or require further assistance, please contact AWS Support [4].

    [1] https://nodejs.org/en/blog/release/v12.22.12/
    [2] https://docs.aws.amazon.com/lambda/latest/dg/runtime-support-policy.html
    [3] https://aws.amazon.com/cli/
    [4] https://aws.amazon.com/support

    Sincerely,
    Amazon Web Services

Many of the lambdas using the NodeJS 12 runtime in my account are generated by CDK constructs/stacks, in particular aws_eks.
May you upgrade to NodeJS 16 ?

Thank you!

Chris

### Expected Behavior

NodeJS above 12

### Current Behavior

NodeJS 12

### Reproduction Steps

deploy an eks cluster using the aws_eks construct

### Possible Solution

Update CDK lambda layer and lambda runtime to a default of >=13. Ability to determine lambda layer runtime via parameter on the aws_eks.Cluster construct could also be very useful.

### Additional Information/Context

Stack deployed using CDK CLI 2.42.0 and Node v 16.13.1 and npm 8.1.2. 

### CDK CLI Version

2.42.0

### Node.js Version

16.13.1

### OS

Ubuntu Linux

### Language

Typescript

### Language Version

4.5.4

**Comment by peterwoodworth** [comment]
Similar to [this issue and comment](https://github.com/aws/aws-cdk/issues/21718#issuecomment-1224600143), this has been addressed a bit ago. Could you please link to any functions in our codebase which are still configured to use node 12? thanks

**Comment by ctrepka** [comment]
@peterwoodworth Thanks for your reply and for linking the other issues. 

After further investigation I found that, although I had updated aws-cdk in the package.json to >=v2.28.0, I had not updated the aws-cdk-lib to >=v2.28.0. After updating both, deleting node_modules and package-lock.json, and rerunning `npm install`, everything worked like a charm and `cdk diff` produced the expected results (lambdas with Node.JS runtime v14)

Thanks again for your help!
